src/compiler/parser.ts

```diff
@@ -440,7 +440,26 @@ namespace ts {
 
     /* @internal */
     export function parseIsolatedJSDocComment(content: string, start?: number, length?: number) {
-        return Parser.JSDocParser.parseIsolatedJSDocComment(content, start, length);
+        const result = Parser.JSDocParser.parseIsolatedJSDocComment(content, start, length);
+        if (result.jsDocComment) {
+            // because the jsDocComment was parsed out of the source file, it might
+            // not be covered by the fixupParentReferences.
+            let parentNode: Node = result.jsDocComment;
+            forEachChild(result.jsDocComment, visitNode);
+
+            function visitNode(n: Node): void {
+                if (n.parent !== parentNode) {

```

Comment:
I think this might happen during incremental parsing; however as the jsDocComment is parsed in isolation, it is always parsed from scratch, so you are right that it should never happen. Will update.